backend/de.metas.async/src/main/java/de/metas/async/api/impl/WorkPackageQueue.java

```diff
@@ -128,13 +130,14 @@ private WorkPackageQueue(@NonNull final Properties ctx,
 		dao = Services.get(IQueueDAO.class);
 
 		this.ctx = ctx;
-		this.packageProcessorIds = Collections.unmodifiableList(new ArrayList<>(packageProcessorIds));
+		this.packageProcessorIds = packageProcessorIds;
+		this.queueProcessorId = queueProcessorId;
 		this.priorityFrom = priorityFrom;
 		this.skipRetryTimeoutMillis = Async_Constants.DEFAULT_RETRY_TIMEOUT_MILLIS;
 
 		if (forEnqueing)
 		{
-			enquingPackageProcessorId = packageProcessorIds.get(0);
+			enquingPackageProcessorId = packageProcessorIds.iterator().next().getRepoId();
```

Comment:
consider changing enquingPackageProcessorId's type from int to QueuePackageProcessorId